ceph_deploy/mon.py

```diff
@@ -536,6 +527,25 @@ def make(parser):
 #
 
 
+def get_mon_initial_members(args, error_on_empty=False, _cfg=None):
+    """
+    Read the ceph config file and return the value of mon_initial_members
+
+    Optionally, a NeedHostError can be raised if the value is None.
+    """
+    if _cfg:
+        cfg = _cfg
+    else:
+        cfg = conf.ceph.load(args)
+    mon_initial_members = cfg.safe_get('global', 'mon_initial_members')
+    if not mon_initial_members and error_on_empty:
+        raise exc.NeedHostError(
+            'could not find `mon initial members` defined in ceph.conf'
+        )
+    monitors = re.split(r'[,\s]+', mon_initial_members)
```

Comment:
I think this is going to throw an exception if error_on_empty is False.  The pre-ceding code will throw an expection if error_on_empty is True, but if error_on_empty is False, `mon_initial_members` could be None.  If it is, this is going to call re.split(...., None), and that will lead to a TypeError.  You can't give None to split()

Instead, this needs to be:

```
if not mon_initial_members:
    if error_on_empty:
         raise...
else:
    mon_initial_members = re.split(r'[,\s]+', mon_initial_members)
return mon_initial_members
```
There is no need for the `monitors` intermediate variable. I feel like it was like that at one point, but got changed.  I could be wrong.